### DISCUSSION TITLE
Alpha Channel problem after profile assignment

### DISCUSSION THREAD

**Discussion by Manu-Sal** [ImageMagick/ImageMagick#2839] [Development]
Hello there, 

I use imageMagick mainly to assign color profiles and set the resolution of my 3D renderings automatically to 300dpi, which basically works fine. Before I use imageMagick, the alpha channel of my image is fine. After usage, the alpha channel is still fine in most graphic software. But if I open it in InDesign, it says transparency = No and the semitranparent pixels of my images appear black. I need to open the picture in photoshop or other similar software again and save is without changing a thing. After that, it is fine in InDesign as well. My goal was to assign the profile and make little changes with imageMagick, so I don't need to manually use a graphic program or anything else.

Here is the command I use for assigning the profile to my *.tif files. I don't change the alpha or anything else. My ImageMagick Version is ImageMagick-7.0.9-Q16:
magick convert imgName -units PixelsPerInch -density 300 -profile eciRGB_v2.icc imgName

I already had a look at maaaaany many online forums and tried different solutions or additional commands like -alpha, -background, -repage, -channel and so on... but nothing solved my problem, I think this might be a bug...?
I would be very happy if anyone could help me solve the problem and I am sure many other people would be happy too ;-)

Thank you very much and nice greetings from Austria!

**Comment by fmw42** [comment]
Try adding one of the options for -define tiff:alpha=associated or unassociated or unspecified just before the tiff output and see if that works. Most likely it would be unspecified. But if that does not work, try the other options.

**Reply by Manu-Sal** [reply]
thanks, one of the options "associated" worked somehow, it resulted in an existing transparency in InDesign - yay! Unfortunately, the opazity of the semitransparent pixels has also changed, as you can see in this picture of my output:
![compare-associated](https://user-images.githubusercontent.com/74615943/99427683-024bd980-2906-11eb-97ec-a719815c5ee0.JPG)

**Reply by fmw42** [reply]
Did you try the others? If so, then I do not know what is happening. One of the IM developers may need to look into this.

**Reply by Manu-Sal** [reply]
Yes, I tried the other options as well, but get no transparency in InDesign. It always looks like in the following image (no transparency in InDesign):
![inDesign](https://user-images.githubusercontent.com/74615943/99428457-f57bb580-2906-11eb-8337-04cf4f79bb14.JPG)

**Comment by fmw42** [comment]
Post your input image and your output image as separate images (perhaps zip them to be sure no changes occur) so the developers can examine them and look into your issue.

**Reply by Manu-Sal** [reply]
Okay good idea, that would be great!
Here is a zip with input and output of the command including your tipp with -define tiff:alpha=associated. 
[comparison-before-after.zip](https://github.com/ImageMagick/ImageMagick/files/5555515/comparison-before-after.zip)

**Comment by snibgo** [comment]
> I need to open the picture in photoshop or other similar software again and save is without changing a thing.

Please post a sample image before and after saving in Photoshop. Then we can see what it changes.

Please also post the text output from "magick -version".

**Reply by Manu-Sal** [reply]
Output from "magick -version":
Version: ImageMagick 7.0.9-14 Q16 x64 2020-01-05 http://www.imagemagick.org
Copyright: Copyright (C) 1999-2018 ImageMagick Studio LLC
License: http://www.imagemagick.org/script/license.php
Visual C++: 180040629
Features: Cipher DPC Modules OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg lcms lqr lzma openexr pangocairo png ps raw rsvg tiff webp xml zlib

New sample images:
[comparison-before-after.zip](https://github.com/ImageMagick/ImageMagick/files/5559888/comparison-before-after.zip)

**Reply by Manu-Sal** [reply]
I think the "assotiated" option leads in the right direction. Maybe someone knows a workaround for the changed opacity of the semitransparent pixels...

**Comment by fmw42** [comment]
As a test for the developers. Try converting again with -define tiff:alpha=unspecified. But then open the file in PS and enable the alpha channel and save the file. Does that open in your other app?

**Reply by fmw42** [reply]
Please note that for best results with IM 7, use magick, not magick convert.

**Reply by Manu-Sal** [reply]
okay, I added this test too in the new sample images above. 
I also tried it with magick instead of magick convert. The output looks equal to me.

**Comment by snibgo** [comment]
Comparing "magick -verbose info:" on your case2 files:

after_imageMagick_before_photoshop.tif
after_imageMagick_after_photoshop.tif

The "before_photoshop" version has alpha "unassociated".

The "after_photoshop" version has two images. In both, alpha is "associated". I suppose you only use the first image??

I can't see other significant differences. So maybe making alpha "associated" will fix your problem, like this:
```
magick after_imageMagick_before_photoshop.tif -define tiff:alpha=associated z.tiff
```
Does z.tiff behave like after_imageMagick_after_photoshop.tif and do what you want?

**Reply by Manu-Sal** [reply]
Unfortunately not, this is what @fmw42 also showed me above. With this option, the transparency shows up in InDesign (which is what I asked for), but the opacity of the pixels has also changed too much with this command, as you can see in the following picture (this is the same picture you can find above). That's why I cannot use it. The output files can also be found in the last zip ("case 1" of 3).
![compare-associated](https://user-images.githubusercontent.com/74615943/99549047-8dd27280-29b9-11eb-970a-79a2695b397a.JPG)

**Comment by fmw42** [comment]
Are you using Gray or Spot dot gain? If so, ImageMagick does not know about those and will skip them.

**Reply by Manu-Sal** [reply]
To be honest, I don't know the answer to your question... I get the input-tif like it is. But your question gives me an idea. I will try the command with another color profile and see if that behaves differently.

**Comment by snibgo** [comment]
after_imageMagick_after_photoshop.tif contains metadata added by Photoshop, including "layer" metadata. IM won't add this metadata, and will ignore it when reading files.

I suppose InDesign is (like Photoshop) an Adobe product, so probably uses that metadata. This might be the problem you face.

You said:
> I need to open the picture in photoshop or other similar software again and save is without changing a thing.

By "other similar software" do you mean "other Adobe software", or do other products (such as Gimp) also fix the problem?

**Reply by Manu-Sal** [reply]
I tried it with Photoshop and Gimp. Here you can find a test output of gimp, which works fine in InDesign afterwards too:
[after_imageMagick_after_gimp.zip](https://github.com/ImageMagick/ImageMagick/files/5564966/after_imageMagick_after_gimp.zip)

A question for understanding it a bit better... If IM doesn't add or read metadata, how can the file be different then after using IM? The "beforeIM" file can be opened without problems in any software, but has no color profile. After adding the profile with IM, I have this problem. Is there maybe another way with IM to add a color profile and set it to 300dpi? My IM command obviously creates the problem with the transparency. And if I use "associated", the transparency works but has changed its opacity.

If that's simply not possible, this answer would also be fine for me.
Thanks again for your help ;-)

**Reply by Manu-Sal** [reply]
Side-Note: The "beforeIM" has never seen photoshop before, as we try to avoid photoshop for this process.

**Comment by fmw42** [comment]
You are adding the color profile and setting the density properly and that is not likely the issue. My suspicion is that it is the xmp profile that IM is not likely using or it is the way IM sets the associated transparency and while doing so changes the base alpha. Perhaps IM is trying to change the alpha to binary rather than keep it 8-bits. **I think the IM developers need to look into this.**

**Comment by motz17** [comment]
I have exactly the same issue.  (using ImageMagick 7.1.1-15 Q16-HDRI x64)
I can confirm that it is not related to any profile assignment.

@Manu-Sal Did you find a solution for this issue?


I did find out, that the argument `-define tiff:rows-per-strip=1 `  solves the problem of changing the base alpha, but then the original problem is present again (INDD background appears black).

I also experimented with other arguments like `-type truecoloralpha`  and `-depth 16` but without success.

**Comment by motz17** [comment]
This is my sample & result: 
[Asset_defineTiffAlpha=associated.zip](https://github.com/ImageMagick/ImageMagick/files/14537929/Asset_defineTiffAlpha.associated.zip)

my used call: 
`magick Asset_orginal.tif[0] -compress lzw -define tiff:alpha=associated -depth 16 Asset_defineTiffAlpha=associated.tif"`

**Comment by coeing** [comment]
Hey everybody,

I am glad to have found this discussion, as we are experiencing the same problem. We have a PNG that we want to convert into a TIFF file that can be used in Adobe InDesign as well.

My findings are:
- Just converting the PNG to TIFF results in a non-transparent background in InDesign (but it is transparent in Gimp, so it is not a pure IM issue)
- Using `-define tiff:alpha=associated` makes the background in InDesign transparent. But it also changes the semi-transparent pixels too much to work with the result.
- In Gimp there is an option "Save color values from transparent pixels" ("Colors are not stored premultiplied by the associated alpha") when exporting to TIFF. If this option is checked, the background is not transparent in InDesign as well. So this seems to work the same way than the IM define. But other than in IM, the semi-transparent pixels are not modified.

![image](https://github.com/user-attachments/assets/45e895f0-f6dc-474e-a9ef-32fd6ac6fe0f)

Here is my test file: 
![original](https://github.com/user-attachments/assets/809bcafb-24d5-49a8-9f9a-e31c03398743)

I am willing to help with solving this issue, so just tag me, I will have an eye on this thread.

@Manu-Sal As this thread is from 2020: Did you find a solution in the mean time? My workaround would be to use Gimp in our pipeline to do the conversion instead, but I would love to avoid having another tool introduced there 🙂

**Reply by fmw42** [reply]
Post your TIFF output from IM and your TIFF output that you get from GIMP as what you desire.  Have you tried all 3 options from -define tiff:alpha=XXX.  Post your exact command line from Imagemagick for your TIFF output from Imagemagick.

**Reply by coeing** [reply]
@fmw42 Yes, I tried them all. `unassociated` and `undefined` will show the background again in InDesign, but the colors look correct. `associated` shows a transparent background, but the semi-transparent pixels are incorrect.

Here is small comparison between IM and Gimp:
![image](https://github.com/user-attachments/assets/6372580d-1e83-449b-a6af-53ae64982c04)

In Gimp:
- Saving the color values on export will also show the background in InDesign
- Not saving the color values will not show the background and will preserve the correct semi-transparent pixels. So this is the result that I wish to achieve.

ImageMagick 7.1.1:
- associated: `magick original.png -define tiff:alpha=associated -compress LZW im-compress-alpha-associated.tiff`
- unassociated: `magick original.png -define tiff:alpha=unassociated -compress LZW im-compress-alpha-unassociated.tiff`
- undefined: `magick original.png -define tiff:alpha=undefined -compress LZW im-compress-alpha-undefined.tiff`

If you need more information, just let me know 🙂

**Reply by coeing** [reply]
Here are the files from IM and Gimp as well:
[gimp-im-comparison.zip](https://github.com/user-attachments/files/16767577/gimp-im-comparison.zip)

**Comment by coeing** [comment]
@Manu-Sal  @motz17 @fmw42 I created a bug report now, maybe an IM developer finds some time to look into this issue: https://github.com/ImageMagick/ImageMagick/issues/7578

**Comment by fmw42** [comment]
There is no difference between the original and unassoc or undefined LZW Tiffs.

```
magick original.png -define tiff:alpha=undefined -compress LZW alpha_undef_lzw.tiff
magick original.png -define tiff:alpha=unassociated -compress LZW alpha_unassoc_lzw.tiff
magick compare -metric rmse original.png alpha_unassoc.tiff null:
6.77417e-13 (1.03367e-17)
magick compare -metric rmse original.png alpha_unassoc_lzw.tiff null:
6.77417e-13 (1.03367e-17)bash-5.2$ 

```
These files are virtually identical.

I am using IM 7.1.1-37 on Mac OSX Ventura with libtiff 4.6.0

**What is your version of libtiff?**
